webapp/src/main/java/com/box/l10n/mojito/android/strings/AndroidStringDocumentWriter.java

```diff
@@ -0,0 +1,145 @@
+package com.box.l10n.mojito.android.strings;
+
+import com.google.common.base.Strings;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+
+import static com.box.l10n.mojito.android.strings.AndroidStringDocumentUtils.TRANSFORMER_FACTORY;
+import static com.box.l10n.mojito.android.strings.AndroidStringDocumentUtils.documentBuilder;
+import static com.box.l10n.mojito.android.strings.AndroidStringElement.ID_ATTRIBUTE_NAME;
+import static com.box.l10n.mojito.android.strings.AndroidStringElement.NAME_ATTRIBUTE_NAME;
+import static com.box.l10n.mojito.android.strings.AndroidStringElement.PLURAL_ELEMENT_NAME;
+import static com.box.l10n.mojito.android.strings.AndroidStringElement.PLURAL_ITEM_ELEMENT_NAME;
+import static com.box.l10n.mojito.android.strings.AndroidStringElement.QUANTITY_ATTRIBUTE_NAME;
+import static com.box.l10n.mojito.android.strings.AndroidStringElement.ROOT_ELEMENT_NAME;
+import static com.box.l10n.mojito.android.strings.AndroidStringElement.SINGULAR_ELEMENT_NAME;
+import static java.util.Objects.requireNonNull;
+
+public class AndroidStringDocumentWriter {
+
+    private AndroidStringDocument source;
+    private DOMSource domSource;
+    private Document document;
+    private Node root;
+
+    public AndroidStringDocumentWriter(final AndroidStringDocument source) throws ParserConfigurationException {
+        this.source = requireNonNull(source);
+        buildDomSource();
+    }
+
+    public void buildDomSource() throws ParserConfigurationException {
+        document = documentBuilder().newDocument();
+        root = document.createElement(ROOT_ELEMENT_NAME);
+        document.setXmlStandalone(true);
+        document.appendChild(root);
+        source.getSingulars().forEach(this::addSingular);
```

Comment:
I think that confirm my previous feeling. For our specific use case we split files singular vs plural but maybe here it make sense to have an implementation a little bit more flexible that respect the order in which the strings are added

Reply:
Updated